### PR TITLE
Update pre-commit-hooks repo version

### PR DESCRIPTION
**Pull request by feluelle**
- use official isort pre-commit-hook
- use official yamllint pre-commit-hook
- run isort pre-commit-hook on all python files instead of files ending with py

---
Make sure to mark the boxes below before creating PR: [x]

- [x] Description above provides context of the change
- [x] Unit tests coverage for changes (not needed for documentation changes)
- [x] Target Github ISSUE in description if exists
- [x] Commits follow "[How to write a good git commit message](http://chris.beams.io/posts/git-commit/)"
- [x] Relevant documentation is updated including usage instructions.
- [x] I will engage committers as explained in [Contribution Workflow Example](https://github.com/apache/airflow/blob/master/CONTRIBUTING.rst#contribution-workflow-example).

---
In case of fundamental code change, Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvements+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in [UPDATING.md](https://github.com/apache/airflow/blob/master/UPDATING.md).
Read the [Pull Request Guidelines](https://github.com/apache/airflow/blob/master/CONTRIBUTING.rst#pull-request-guidelines) for more information.

### CODE REVIEW

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -141,7 +141,7 @@ metastore_browser/templates/.*\\.html$|.*\\.jinja2"
     hooks:
       - id: check-hooks-apply
   - repo: https://github.com/pre-commit/pre-commit-hooks
-    rev: v2.5.0
+    rev: v3.1.0
```
Comment (feluelle) [inline .pre-commit-config.yaml:144]: > pre-commit/pre-commit-hooks now requires python3.6.1+

That is the only noticeable difference for us.

Ref: https://github.com/pre-commit/pre-commit-hooks/releases

File: .pre-commit-config.yaml [hunk 2]
```diff
@@ -159,15 +159,23 @@ metastore_browser/templates/.*\\.html$|.*\\.jinja2"
     hooks:
       - id: rst-backticks
       - id: python-no-log-warn
-  - repo: local
+  - repo: https://github.com/adrienverge/yamllint
+    rev: v1.23.0
```
Comment (feluelle) [inline .pre-commit-config.yaml:163]: Fix version - I set it to the latest version.

File: .pre-commit-config.yaml [hunk 3]
```diff
@@ -159,15 +159,23 @@ metastore_browser/templates/.*\\.html$|.*\\.jinja2"
...
     hooks:
       - id: yamllint
         name: Check yaml files with yamllint
         entry: yamllint -c yamllint-config.yml
-        language: python
-        additional_dependencies: ['yamllint']
         types: [yaml]
         exclude: ^.*init_git_sync\.template\.yaml$|^.*airflow\.template\.yaml$
+  - repo: https://github.com/timothycrosley/isort
+    rev: 4.3.21-2
```
Comment (feluelle) [inline .pre-commit-config.yaml:171]: Fix version - I set it to the latest version.

**Review by potiuk — APPROVED** [review]

**Comment by potiuk** [comment]
It looks like the new stylelint requires some style improvements as well @feluelle

**Review by turbaszek — APPROVED** [review]

**Comment by ashb** [comment]
isort seems to be wreaking havok with the UPDATING.md -- we might have to not pick the latest version (or else turn it tune it down):

```diff
diff --git a/docs/howto/operator/gcp/gcs_to_gcs.rst b/docs/howto/operator/gcp/gcs_to_gcs.rst
index e17bedf..bb3fda1 100644
--- a/docs/howto/operator/gcp/gcs_to_gcs.rst
+++ b/docs/howto/operator/gcp/gcs_to_gcs.rst
@@ -40,11 +40,9 @@ perform this task faster and more economically. The economic effects are especia
 Airflow is not hosted in Google Cloud Platform, because these operators reduce egress traffic.
 
 These operators modify source objects if the option that specifies whether objects should be deleted
-from the source after they are transferred to the sink is enabled.
 
 When you use the Google Cloud Data Transfer service, you can specify whether overwriting objects that already exist in
 the sink is allowed, whether objects that exist only in the sink should be deleted, or whether objects should be deleted
-from the source after they are transferred to the sink.
 
 Source objects can be specified using include and exclusion prefixes, as well as based on the file
 modification date.
```

It seems to think _any_ line starting with `from` is an import line! We need to only have it operate on .py files

**Comment by feluelle** [comment]
Thanks all - will fix the issues.

**Comment by feluelle** [comment]
> It seems to think any line starting with from is an import line! We need to only have it operate on .py files

This is fixed in isort `develop` but not yet released :/ So I added it here.

**Review by kaxil — APPROVED** [review]

**Pull request merged**